Revert part of e317943c125d7fdbe62c750324640eb30c88fa15 to restore 
AllowEmptyInput default of False unless specified in xorg.conf until
HAL input device support is integrated into HAL for OpenSolaris.

diff -urp -x '*~' -x '*.orig' hw/xfree86/common/xf86Config.c hw/xfree86/common/xf86Config.c
--- hw/xfree86/common/xf86Config.c	2009-05-08 17:41:54.405734000 -0700
+++ hw/xfree86/common/xf86Config.c	2009-05-08 17:42:02.451242000 -0700
@@ -1007,9 +1007,9 @@ configServerFlags(XF86ConfFlagsPtr flags
     }
 #endif
 
-    /* AllowEmptyInput is automatically true if we're hotplugging */
-    xf86Info.allowEmptyInput = (xf86Info.autoAddDevices && xf86Info.autoEnableDevices);
-    xf86GetOptValBool(FlagOptions, FLAG_ALLOW_EMPTY_INPUT, &xf86Info.allowEmptyInput);
+    xf86Info.allowEmptyInput = FALSE;
+    if (xf86GetOptValBool(FlagOptions, FLAG_ALLOW_EMPTY_INPUT, &value))
+	xf86Info.allowEmptyInput = TRUE;
 
     /* AEI on? Then we're not using kbd, so use the evdev rules set. */
 #ifdef XKB
